server/src/main/java/org/opensearch/cluster/health/ClusterStateHealth.java

```diff
@@ -147,6 +150,11 @@ public ClusterStateHealth(final StreamInput in) throws IOException {
         unassignedShards = in.readVInt();
         numberOfNodes = in.readVInt();
         numberOfDataNodes = in.readVInt();
+        if (in.getVersion().onOrAfter(Version.V_1_0_0)) {
+            hasDiscoveredMaster = in.readBoolean();
+        } else {
+            hasDiscoveredMaster = true;
```

Comment:
Not sure I understand this correctly.
I know we have to initialize `hasDiscoveredMaster` but we are saying any version older than OpenSearch 1.0.0 has always discovered a master, even though that might not be true?

Reply:
The check is in place for backward compatibility since the older versions don't have this attribute in their Transport layer payload stream. 
Also, in the older versions, this value is not being populated in the API response, hence customers would not see this value either and they return 503 in case of master not discovered meaning this would not be invoked if master was not discovered.

Reply:
Thats a good point, thanks for helping me understand.